watertap/unit_models/ion_exchange_0D.py

```diff
@@ -0,0 +1,1346 @@
+                ) / (b.resin_eq_capacity / b.resin_max_capacity * (1 - b.c_norm[j]))
+            else:
+                return Constraint.Skip
+
+        @self.Constraint(doc="Resin equilibrium concentration bounds")
+        def eq_resin_eq_capacity(b):
+            return b.resin_eq_capacity <= b.resin_max_capacity
+
+        @self.Constraint(doc="Interstitial velocity")
+        def eq_vel_inter(b):
+            return b.vel_inter == b.vel_bed / b.bed_porosity
+
+        @self.Constraint(doc="Resin surface area per vol")
+        def eq_resin_surf_per_vol(b):
+            return b.resin_surf_per_vol == (6 * (1 - b.bed_porosity)) / b.resin_diam
+
+        @self.Constraint(doc="Contact time")
+        def eq_t_contact(b):
+            return b.t_contact == b.bed_depth / b.vel_inter
+
+        @self.Constraint(doc="Flow through bed constraint")
+        def eq_bed_flow(b):
+            prop_in = b.properties_in[0]
+            return (b.bed_depth * b.bed_porosity) / b.vel_bed == (
+                b.bed_porosity * b.bed_vol
+            ) / prop_in.flow_vol_phase["Liq"]
+
+        # =========== DIMENSIONLESS ===========
+
+        @self.Constraint(doc="Reynolds number")
+        def eq_Re(b):
+            prop_in = b.properties_in[0]
+            return b.Re == (b.vel_bed * b.resin_diam) / prop_in.visc_k_phase["Liq"]
+
+        @self.Constraint(ion_set, doc="Schmidt number")
+        def eq_Sc(b, j):
+            prop_in = b.properties_in[0]
+            return (
+                b.Sc[j]
+                == prop_in.visc_k_phase["Liq"] / prop_in.diffus_phase_comp["Liq", j]
+            )
+
+        @self.Constraint(ion_set, doc="Sherwood number")
+        def eq_Sh(b, j):
+            return (
+                b.Sh[j]
+                == b.Sh_A / b.bed_porosity * b.Re**b.Sh_exp * b.Sc[j] ** b.Sh_exp
+            )
+
+        @self.Constraint(doc="Bed Peclet number")
+        def eq_Pe_bed(b):
+            return b.Pe_bed == b.Pe_p * (b.bed_depth / b.resin_diam)
+
+        @self.Constraint(doc="Particle Peclet number")
+        def eq_Pe_p(b):
+            return b.Pe_p == b.Pe_p_A * b.Re**b.Pe_p_exp
+
+        @self.Constraint(doc="Service flow rate")
+        def eq_sfr(b):
+            prop_in = b.properties_in[0]
+            return (
+                b.sfr
+                == pyunits.convert(
+                    prop_in.flow_vol_phase["Liq"], to_units=pyunits.m**3 / pyunits.hr
+                )
+                / b.bed_vol
+            )
+
+        @self.Constraint(doc="Bed area calculation")
+        def eq_bed_area(b):
+            return b.bed_area == Constants.pi * (b.bed_diam / 2) ** 2
+
+        @self.Constraint(doc="Bed volume calculation")
+        def eq_bed_vol(b):
+            return self.bed_vol == self.bed_area * self.bed_depth
+
+        @self.Constraint(doc="Bed depth to bed diameter ratio")
+        def eq_bed_depth_to_diam_ratio(b):
+            return b.bed_depth / b.bed_diam >= b.bed_depth_to_diam_ratio
```

Comment:
What is this, why do we need it, and why is it an inequality? How would the depth-to-diam ratio ever be less than the ratio of the depth to the diam?

Reply:
The ratio of the bed depth to diameter needs to greater than some value (I have seen >2.5-5) so the hydrodynamics of the column are such that assumptions of the model hold and it results in a valid design. If `bed_depth_to_diam_ratio` is a variable, maybe should be renamed to `min_bed_depth_to_diam_ratio`.